src/Features/Core/Portable/Completion/AbstractCompletionService.cs

```diff
@@ -277,12 +278,29 @@ private static CompletionItem Disambiguate(CompletionItem item, CompletionItem e
             CancellationToken cancellationToken)
         {
             var context = new CompletionListContext(document, position, triggerInfo, options, cancellationToken);
-
-            await provider.ProduceCompletionListAsync(context).ConfigureAwait(false);
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (!IsPositionInsideRoot(root, position))
+            {
+                try
+                {
+                    var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                    throw new InvalidOperationException(
+                        message: $"Position '{position}' is not contained in SyntaxTree Span '{root.FullSpan}' source text: {sourceText}");
```

Comment:
i'm wary about this.  you're explicitly putting hte full text of the file into our dump.  i'm not sure that's ok.